com.ibm.streamsx.topology/com.ibm.streamsx.topology.functional.python/Aggregate/Aggregate_cpp.cgt

```diff
@@ -128,25 +130,31 @@ void MY_OPERATOR::process(Tuple const & tuple, uint32_t port)
 	      SplpyGIL lock; 
               python_value = pySplValueToPyObject(value);
 
-	      // Create a tuple to pass to the depickling wrapper function.	  
-	      PyObject * pyTuple;
-              pyTuple = PyTuple_New(2);
-              PyTuple_SET_ITEM(pyTuple, 0, python_value);
-              Py_INCREF(python_value);
-              PyTuple_SET_ITEM(pyTuple, 1, python_value);
-
 	      // Depickle the tuple.
-	      python_value = streamsx::topology::SplpyGeneral::pyCallObject(spl_in_object_out, pyTuple);
+	      PyObject *tup = PyTuple_New(1);
+	      PyTuple_SET_ITEM(tup, 0, python_value);
+   	      python_value = SplpyGeneral::pyCallObject(loads, tup);
+
 	  } // End SplpyGIL lock
       }
-  <%} else {%>
-      	{
-	    // If the tuple is not a pickled object, or a reference, convert it to a PyObject
-	    // using pySplProcessTuple
-	    SplpyGIL lock;
-            python_value = streamsx::topology::pySplProcessTuple(spl_in_object_out, value);
-	} // End SplpyGIL lock
-  <%}%>
+  <% } elsif ($pystyle eq 'string'){%>
+      {
+	  SplpyGIL lock;
+          python_value = pySplValueToPyObject(value);
+      }
+  <% } elsif ($pystyle eq 'json'){%>

```

Comment:
Minor point, but aren't the pickle and json code paths the same now? So they could be grouped together as there are elsewhere in this PR?